src/main/java/org/opentripplanner/routing/algorithm/filterchain/ItineraryFilterChainBuilder.java

```diff
@@ -97,6 +99,15 @@ public ItineraryFilterChainBuilder withTransitGeneralizedCostLimit(DoubleFunctio
         this.transitGeneralizedCostLimit = value;
         return this;
     }
+
+    /**
+     * This is the same as {@link #withTransitGeneralizedCostLimit(DoubleFunction)}, only for
+     * non-transit itineraries.
+     */
```

Comment:
This is the man documentation for this filter. Describe what it does, witch itineraries it apply to, and witch itineraries is used to find the minimum cost. It is nice to point out that it is similar to the `TransitGeneralizedCostLimit`, but the doc needs to "stand on its own feet" - what if we delete the other filter?

This doc is targeting developers, almost the same doc can be copied to the `Configuration.md`.